packages/Webkul/Product/src/Models/Product.php

```diff
@@ -500,25 +500,25 @@ public function getCustomAttributeValue($attribute)
 
         if ($attribute->value_per_channel) {
             if ($attribute->value_per_locale) {
-                $attributeValue = $this->attribute_values()
+                $attributeValue = $this->attribute_values
```

Comment:
This can't be `attribute_values` directly as this will load the data.

Reply:
This will load for each call a new relation with a database query, is it designed to work in this way?
In this case u will use the cached value

Reply:
In the old one, the motive is to not load unless and until the full query is made.

But in this case, it will load the values on each call.

Reply:
@devansh-webkul But doing so saves queries, reducing loading times. For example, when a product has many variants or sot. What problems can it generate?
As soon as I have a moment, I will go ahead and update the repositories to enable the use of caching.

Reply:
I will discuss it with the team and let you know.

Reply:
Actually, I am just worried about the removal of the query builder. Because there are some products in which there are lots of attributes this will make a huge impact.

Reply:
Have you tested this in your instance?

Reply:
> Actually, I am just worried about the removal of the query builder. Because there are some products in which there are lots of attributes this will make a huge impact.
The attribute model is pretty lightweight, so a collection can store about 4/5k models without any worries.
But this could be a right worried in some cases, maybe we can pre-check how many attributes are present in the database, cache the result, and use a configuration to set a threshold to prevent the cached attribute logic and use the query directly.
We will investigate to find out a possible threshold value

Reply:
Actually, there are some products that have 200+ attributes. And also product counts are up to 70K plus which is the reason I am taking time on this.

Reply:
Get it, we will do some benchmark and test it

Reply:
And really sorry for the late reply. I am just busy with some other projects.